gateway-ha/src/main/java/io/trino/gateway/proxyserver/ProxyRequestHandler.java

```diff
@@ -171,34 +174,39 @@ private void performRequest(
 
         if (statementPaths.stream().anyMatch(request.getUri().getPath()::startsWith) && request.getMethod().equals(HttpMethod.POST)) {
             future = future.transform(response -> recordBackendForQueryId(request, response), executor);
+            if (includeClusterInfoInResponse) {
+                oauth2GatewayCookie.add(new NewCookie.Builder("trinoClusterHost").value(remoteUri.getHost()).build());
+            }
         }
 
+        ImmutableList<NewCookie> cookie = ImmutableList.copyOf(oauth2GatewayCookie);
```

Comment:
alter `buildResponse` to accept a `List` instead of copying `oauth2GatewayCookie` to a new `ImmutableList`.